### DISCUSSION TITLE
How can I access path params from router.subscribe?

### DISCUSSION THREAD

**Discussion by jmorey28** [TanStack/router#2639] [Q&A]
We have analytics events setup on router.subscribe, but we need to be able to access path params so we can send them up separately from the path. As far as I can tell path params do not exist on event.toLocation.

**Comment by schiller-manuel** [comment]
you can access `router.state.matches` which carries path params per match
=> https://tanstack.com/router/latest/docs/framework/react/api/router/RouteMatchType
